lib/Util.php

```diff
@@ -52,11 +54,44 @@ public static function deleteUser(IUser $user) {
 	}
 
     public static function getTurnSettings(IConfig $config, $uid) {
-        $value = $config->getUserValue($uid, 'spreed', 'turn_settings');
-        if (empty($value)) {
-            return array();
+		$value = $config->getUserValue($uid, 'spreed', 'turn_settings');
+		if (empty($value)) {
+			return array();
         }
-        return json_decode($value, true);
+		return json_decode($value, true);
     }
 
+	/**
+	 * Generates a username and password for the TURN server based on the
+	 *
+	 * @param IConfig $config
+	 * @param ISession $session
+	 * @param ITimeFactory $timeFactory
+	 * @return array
+	 */
+    public static function generateTurnSettings(IConfig $config, ISession $session, ITimeFactory $timeFactory) {
+		// generate from shared secret
+		$turnServer = $config->getAppValue('spreed', 'turn_server', '');
+		$turnServerSecret = $config->getAppValue('spreed', 'turn_server_secret', '');
+		$turnServerProtocols = $config->getAppValue('spreed', 'turn_server_protocols', '');
+		$sessionId = $session->get('spreed-session');
+
+		if ($turnServer === '' || $turnServerSecret === '' || $turnServerProtocols === '' || empty($session)) {
+			return array();
+		}
+
+		$time = $timeFactory->getTime();
+
+		$string = sprintf('%d:%s', $time + 3600, $sessionId);
```

Comment:
The TTL should probably a parameter and must also be returned to the user, so it can refresh the credentials before they expire (which can happen if the user is connected longer than the TTL).